lib/cylc/scheduler_cli.py

```diff
@@ -88,6 +88,12 @@ def main(is_restart=False):
         reg = SuiteSrvFilesManager().register()
         # Replace this process with "cylc run REG ..." for easy identification.
         os.execv(sys.argv[0], [sys.argv[0]] + [reg] + sys.argv[1:])
+    elif not os.path.exists(
+            SuiteSrvFilesManager().get_suite_srv_dir(args[0], options.owner)):
+        # Source path is assumed to be the run directory
+        SuiteSrvFilesManager().register(
+            args[0],
+            glbl_cfg().get_derived_host_item(args[0], 'suite run directory'))
```

Comment:
This block has to go after the host appointer logic.

I am unable to get tests/authentication/10 to work in my environment otherwise. (The test attempts to start up a suite on a remote host without a shared file system - so we cannot register a suite until we are already running on the assigned remote host.)

Reply:
Alternatively, we should just delete tests/authentication/10. Do we really need to have this functionality?